### PR TITLE
Sorting the collections

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

File: pkg/collections.go [hunk 1]
```diff
@@ -260,38 +288,69 @@ func (c *copier) copyIndexes(ctx context.Context, sourceColl driver.Collection,
...
+		ShardKeys:         props.ShardKeys,
+		NumberOfShards:    props.NumberOfShards,
+		IsSystem:          false,
+		Type:              props.Type,
+		KeyOptions: &driver.CollectionKeyOptions{
+			AllowUserKeys: props.KeyOptions.AllowUserKeys,
+			Type:          props.KeyOptions.Type,
+		},
+		DistributeShardsLike: props.DistributeShardsLike,
+		IsSmart:              false,
```
Comment (ewoutp) [inline pkg/collections.go:321]: Shoudn't we copy this?

File: pkg/collections.go [hunk 2]
```diff
@@ -260,38 +288,69 @@ func (c *copier) copyIndexes(ctx context.Context, sourceColl driver.Collection,
...
+
+// sortCollections sorts a collection list on the following premis:
+// No DistributedShardsLike comes before DistributedShardsLike. In case two collections don't have this setting
+// a vertex collection comes before an edge collection.
+func (c *copier) sortCollections(collections []driver.Collection, m map[string]driver.CollectionProperties) error {
+	sort.SliceStable(collections, func(i, j int) bool {
+		pi := m[collections[i].Name()]
+		pj := m[collections[j].Name()]
+		if pi.DistributeShardsLike == "" && pj.DistributeShardsLike != "" {
+			return true
```
Comment (ewoutp) [inline pkg/collections.go:349]: Add comments explaining all possible paths

File: pkg/collections.go (hunk 1 above)
Comment (Skarlso) [reply pkg/collections.go:321]: Actually, looking at go-driver, there seems to be something wrong. The API does not return isSmart. But it also does not return SmartJoinAttribute it returns SmartGraphAttribute ( for a collection! ). I need to talk to Adam to verify if this is a bug in go-driver or a bug in arango-api.

Comment (Skarlso) [reply pkg/collections.go:321]: Ok, so two things. The API doc is wrong... Second, IsSmart is actually not used in go-driver... :) Also, not returned by the API at all so it's just an ignored field basically in the create Options.

I'm still setting smartJoinbAttribute though. I'll talk with Adam if we should remove it all together.

**Review by ewoutp — APPROVED** [review]
LGTM

**Pull request merged**